src/AbstractServiceProvider.php

```diff
@@ -70,4 +74,77 @@ protected function makeClient(Application $app)
 
         return $client;
     }
+
+    /**
+     * Setup user detection.
+     *
+     * @param \Bugsnag\Client    $client
+     * @param \Silex\Application $app
+     *
+     * @return void
+     */
+    protected function setupUserDetection(Client $client, Application $app)
+    {
+        try {
+            $tokens = $app['security.token_storage'];
+            $checker = $app['security.authorization_checker'];
+        } catch (InvalidArgumentException $e) {
+            return;
+        }
+
+        $client->registerCallback(new CustomUser(function () use ($tokens, $checker) {
+            $token = $tokens->getToken();
+
+            if (!$token || !$checker->isGranted('IS_AUTHENTICATED_REMEMBERED')) {
+                return;
+            }
+
+            $user = $token->getUser();
+
+            if ($user instanceof UserInterface) {
+                return ['id' => $user->getUsername()];
+            }
+
+            return ['id' => (string) $user];
+        }));
+    }
+
+    /**
+     * Setup the client paths.
+     *
+     * @param \Bugsnag\Client $client
+     * @param string|null     $strip
+     * @param string|null     $project
+     *
+     * @return void
+     */
+    protected function setupPaths(Client $client, $strip, $project)
+    {
+        if ($strip) {
+            $client->setStripPath($strip);
+
+            if (!$project) {
+                $client->setProjectRoot("{$strip}/src");
+            }
+
+            return;
+        }
+
+        if ($project) {
+            if ($base && substr($project, 0, strlen($base)) === $base) {
+                $client->setStripPath($base);
+            }
+
+            $client->setProjectRoot($project);
+
+            return;
+        }
+
+        $base = realpath(__DIR__.'/../../../../');
```

Comment:
Yeh, lol. SIlex has no way to know the app path, which is why I didn't add this originally, but Simon said it would be nice to have this working if we could, so I went with this.

This works on the assumption that the structure is `vendor/bugsnag/bugsnag-silex/src/AbstractServiceProvider.php`, and then the rest of the project is in the same directory as `vendor`, including an `src` folder with the main app code in it.

The nice thing about realpath is it will return false if the path doesn't exist, so if someone has messed with the directories, then we should be ok. :)